### PR TITLE
[feat/LIVE-9136]: add api and helper hooks for v5 swap transition

### PR DESCRIPTION
**Pull request by kallen-ledger**
### 📝 Description

* add `useAPI` hook to LLC. This is similar to react query's `useQuery` hook.
* add hooks for `/to`, `/from` and `/rates`

### ❓ Context

- **Impacted projects**: `LLC` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

N/A

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/hooks/useAPI.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        setData(cachedItem.data);
+        onSuccess?.(cachedItem.data);
+        setIsLoading(false);
+      } else {
+        // The cached item is stale or doesn't exist, so we fetch fresh data
+        fetch();
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [cacheKey, staleTimeout, enabled]);
```
Comment (sarneijim) [inline libs/ledger-live-common/src/hooks/useAPI.ts:75]: Being a callback, can we use all the deps and remove this comment?

File: libs/ledger-live-common/src/exchange/swap/types.ts [hunk 2]
```diff
@@ -34,6 +34,35 @@ export type ExchangeRate = {
   expirationTime?: number;
 };
 
+type ExchangeRateV5ProviderType = "DEX" | "CEX" | "DISABLED";
```
Comment (sarneijim) [inline libs/ledger-live-common/src/exchange/swap/types.ts:37]: Are we receiving DISABLED type from backend?

**Review by sarneijim — APPROVED** [review]
Changeset? 
https://giphy.com/clips/viralhog-viral-hog-otter-claps-and-taps-for-clams-vfI3uwl4FMuDaLbZah

File: libs/ledger-live-common/src/exchange/swap/types.ts (hunk 2 above)
Comment (kallen-ledger) [reply libs/ledger-live-common/src/exchange/swap/types.ts:37]: This was defined on the swagger file. But I've yet to see it.

**Pull request merged**